core/src/main/java/hudson/model/Executor.java

```diff
@@ -843,7 +845,13 @@ public HttpResponse doStop() {
         lock.writeLock().lock(); // need write lock as interrupt will change the field
         try {
             if (executable != null) {
-                Tasks.getOwnerTaskOf(getParentOf(executable)).checkAbortPermission();
+                final SubTask parentOf;
+                try {
+                   parentOf = getParentOfOrFail(executable);
+                } catch(InvocationTargetException ex) {
+                    return HttpResponses.error(500, ex);
```

Comment:
This is a non-Restricted public API (it should be restricted, of course), hence I would prefer to keep the current implementation if you do not feel strongly. The user-visible behavior does not change